rir/src/api.cpp

```diff
@@ -210,35 +212,30 @@ REXPORT SEXP pir_tests() {
     return R_NilValue;
 }
 
-// startup ---------------------------------------------------------------------
+SEXP pirOptDefaultOpts(SEXP closure, SEXP name) {
+    std::string n = "";
+    if (TYPEOF(name) == SYMSXP)
+        n = CHAR(PRINTNAME(name));
+    // PIR can only optimize closures, not expressions
+    if (isValidClosureSEXP(closure) && DispatchTable::check(BODY(closure)))
```

Comment:
@mhyee pointed me in a previous PR that `DispatchTable::check` already test if it is a valid closure...

Reply:
it doesn't

Reply:
the other way round, isValidClosureSEXP checks if it's a dispatch table in the closure